src/datanode/src/sql/alter.rs

```diff
@@ -43,12 +44,29 @@ impl SqlHandler {
                 table_name: &full_table_name,
             }
         );
-        self.table_engine
-            .alter_table(&ctx, req)
-            .await
-            .context(error::AlterTableSnafu {
-                table_name: full_table_name,
-            })?;
+        let kind = req.alter_kind.clone();
+        let table =
+            self.table_engine
+                .alter_table(&ctx, req)
+                .await
+                .context(error::AlterTableSnafu {
+                    table_name: full_table_name,
+                })?;
+        let table_info = &table.table_info();
+        if let AlterKind::RenameTable { .. } = kind {
```

Comment:
You could add an `is_rename_table()` method to the `AlterKind`, then we don't need to clone it here. By the way, we only need to acquire the table_info when the alter kind is rename.
```suggestion
        let is_rename = req.is_rename_table();
        let table =
            self.table_engine
                .alter_table(&ctx, req)
                .await
                .context(error::AlterTableSnafu {
                    table_name: full_table_name,
                })?;
        if is_rename {
            let table_info = &table.table_info();
```